pulp_2_tests/tests/docker/api_v2/test_remove.py

```diff
@@ -50,22 +197,60 @@ def setUp(self):
                 'enable_v1': False,
                 'enable_v2': True,
                 'feed': DOCKER_V2_FEED_URL,
-                'upstream_name': get_upstream_name(self.cfg),
+                'upstream_name': DOCKER_UPSTREAM_NAME,
             }
         )
         self.repo = self.client.post(REPOSITORY_PATH, body)
         self.addCleanup(self.client.delete, self.repo['_href'])
         sync_repo(self.cfg, self.repo)
 
+    def create_and_copy_test_repo(self, source_repo, copy_units):
+        """Return repo of repos for test."""
+        body = gen_repo(
+            importer_config={
+                'enable_v1': False,
+                'enable_v2': True,
+            }
+        )
+        repo = self.client.post(REPOSITORY_PATH, body)
+        self.copy_docker_units(repo, source_repo, copy_units)
+        return repo
+
+    def copy_docker_units(self, repo, source_repo, units):
+        """Copy specified docker units."""
+        criteria = {
+            'type_ids': [
+                'docker_tag',
+                'docker_manifest_list',
+                'docker_manifest',
+                'docker_blob',
```

Comment:
Do we need to pass all `type_ids`? Can we pass criteria as a function parameter?

Reply:
While I am going to make this a variable where I pass in all the `type_ids`, I don't see the benefit in making `copy` or `delete` functions more complex with additional parameterization with a local function.

Since we are only deleting by specific `sha256`, only the `type_id` matching that unit will get deleted. 

If there as a need to pass a list of parameters, that would have to be determined every time. The possibility of a mis-match on `sha256` and `type_id` doesn't seem worth the effort due to scope of use.